web/src/lib/components/asset-viewer/video-remote-viewer.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+  import CircleIconButton from '$lib/components/elements/buttons/circle-icon-button.svelte';
+  import Icon from '$lib/components/elements/icon.svelte';
+  import LoadingSpinner from '$lib/components/shared-components/loading-spinner.svelte';
+  import { castManager, CastState } from '$lib/managers/cast-manager.svelte';
+  import { mdiCastConnected, mdiPause, mdiPlay } from '@mdi/js';
+  import { t } from 'svelte-i18n';
+
+  interface Props {
+    poster: string;
+    assetFileUrl: string;
+    onVideoStarted: () => void;
+    onVideoEnded: () => void;
+  }
+
+  let { poster, assetFileUrl, onVideoEnded, onVideoStarted }: Props = $props();
+
+  let previousPlayerState: CastState | null = $state(null);
+
+  const handlePlayPauseButton = () => {
+    if (castManager.castState === CastState.PLAYING) {
+      castManager.pause();
+    } else if (castManager.castState === CastState.IDLE) {
+      void cast(assetFileUrl, true);
+    } else {
+      castManager.play();
+    }
+  };
+
+  $effect(() => {
+    if (assetFileUrl) {
+      void cast(assetFileUrl);
+    }
+  });
+
+  $effect(() => {
+    if (
+      castManager.castState !== previousPlayerState &&
+      castManager.castState === CastState.IDLE &&
+      previousPlayerState !== CastState.PAUSED
+    ) {
+      onVideoEnded();
+    }
+
+    previousPlayerState = castManager.castState;
+  });
+
+  const cast = async (url: string, force: boolean = false) => {
+    if (!url) {
+      return;
+    }
+    const fullUrl = new URL(url, globalThis.location.href);
+    const didCast = await castManager.loadMedia(fullUrl.href, force);
+
+    if (didCast) {
+      onVideoStarted();
+    }
+  };
+
+  function handleSeek(event: Event) {
+    const newTime: number = Number.parseFloat((event.target as HTMLInputElement).value);
```

Comment:
```suggestion
    const newTime = Number.parseFloat((event.target as HTMLInputElement).value);
```
I would expect the typing of `parseFloat` to be correct?